deeprank/features/atomicfeature_1.py

```diff
@@ -0,0 +1,703 @@
+            matrix_elec[iA, ind_matrix] = ec
+            matrix_vdw[iA, ind_matrix] = evdw
+
+            # store in the dicts
+            electro_data[keyA] = [np.sum(ec)]
+            vdw_data[keyA] = [np.sum(evdw)]
+
+            # store in the xyz dict
+            key = tuple([0] + xyz[iA, :].tolist())
+            electro_data_xyz[key] = [np.sum(ec)]
+            vdw_data_xyz[key] = [np.sum(evdw)]
+
+            # print the result
+            if _save_ or print_interactions:
+
+                for iB, indexB in enumerate(iB):
+
+                    line = ''
+                    keyB = tuple(atinfo[indexB])
+
+                    line += '{:<3s}'.format(keyA[0])
+                    line += '\t{:>1d}'.format(keyA[1])
+                    line += '\t{:>4s}'.format(keyA[2])
+                    line += '\t{:^4s}'.format(keyA[3])
+
+                    line += '\t{:<3s}'.format(keyB[0])
+                    line += '\t{:>1d}'.format(keyB[1])
+                    line += '\t{:>4s}'.format(keyB[2])
+                    line += '\t{:^4s}'.format(keyB[3])
+
+                    line += '\t{: 6.3f}'.format(r[iB])
+                    line += '\t{: f}'.format(ec[iB])
+                    line += '\t{: e}'.format(evdw[iB])
+
+                    # print and/or save the interactions
+                    if print_interactions:
+                        print(line)
+
+                    if _save_:
+                        line += '\n'
+                        f.write(line)
+
+        # print the total interactions
+        if print_interactions or _save_:
+            line = '\n\n'
+            line += 'Total Evdw  = {:> 12.8f}\n'.format(evdw_tot)
+            line += 'Total Eelec = {:> 12.8f}\n'.format(ec_tot)
+            if print_interactions:
+                print(line)
+            if _save_:
+                f.write(line)
+
+        # close export file
+        if _save_:
+            f.close()
+            print(f'AtomicFeature coulomb and vdw exported to file {fname}')
+
+
+        # add the electrosatic feature
+        self.feature_data['coulomb'] = electro_data
+        self.feature_data_xyz['coulomb'] = electro_data_xyz
+
+        # add the vdw feature
+        self.feature_data['vdwaals'] = vdw_data
+        self.feature_data_xyz['vdwaals'] = vdw_data_xyz
+
+  
+
+    @staticmethod
+    def _prefactor_vdw(r):
+        """prefactor for vdw interactions."""
+
+        r_off, r_on = 8.5, 6.5
+        r2 = r**2
+        pref = (r_off**2 - r2)**2 * (r_off**2 - r2 - 3 *
+                                     (r_on**2 - r2)) / (r_off**2 - r_on**2)**3
+        pref[r > r_off] = 0.
+        pref[r < r_on] = 1.0
+        return pref
+
+
+########################################################################
+#
+#   THE MAIN FUNCTION CALLED IN THE INTERNAL FEATURE CALCULATOR
+#
+########################################################################
+
+def __compute_feature__(pdb_data, featgrp, featgrp_raw, mutant):
```

Comment:
Is there a way to test this feature computation function. We want to be sure it raises no errors and outputs the values in a range that we would expect. If you have such a script, could you include it with the pull please?

Reply:
I would need your help testing this function.